components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCRequestObjectUtil.java

```diff
@@ -105,29 +115,59 @@ private static void validateRequestObjectSignature(OAuth2Parameters oAuth2Parame
         try {
             oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
         } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            OAuth2LogsUtil.log(null, "FAILED", "Server error occurred.", "validate-request-object-signature", null);
             throw new RequestObjectException("Error while retrieving app information for client_id: " + clientId +
                     ". Cannot proceed with signature validation", e);
         }
 
-        // Check whether request object signature validation is enforced.
-        if (oAuthAppDO.isRequestObjectSignatureValidationEnabled()) {
-            if (log.isDebugEnabled()) {
-                log.debug("Request Object Signature Verification enabled for client_id: " + clientId);
-            }
-            if (requestObject.isSigned()) {
-                validateSignature(oAuth2Parameters, requestObject, requestObjectValidator);
+        try {
+            // Check whether request object signature validation is enforced.
+            if (oAuthAppDO.isRequestObjectSignatureValidationEnabled()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Request Object Signature Verification enabled for client_id: " + clientId);
+                }
+                if (requestObject.isSigned()) {
+                    validateSignature(oAuth2Parameters, requestObject, requestObjectValidator);
+                } else {
+                    // If request object is not signed we need to throw an exception.
+                    if (OAuth2LogsUtil.isDiagnosticLogsEnabled()) {
+                        Map<String, Object> params = new HashMap<>();
+                        params.put("clientId", clientId);
+
+                        Map<String, Object> configs = new HashMap<>();
+                        configs.put("requestObjectSignatureValidationEnabled", "true");
+                        OAuth2LogsUtil.log(params, "FAILED",
```

Comment:
define "FAILED" as constant and re-use

Reply:
addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1737/commits/26ac12223c14d5da6faf1a81dffa97cc14eee7bc